src/main/java/org/wildfly/security/util/CodePointIterator.java

```diff
@@ -938,6 +940,122 @@ public static CodePointIterator ofLatin1Bytes(final byte[] bytes, final int offs
     }
 
     /**
+     * Get a sub-iterator that removes the following code points: <code>10</code>(\n) and <code>13</code>(\r).
+     *
+     * @return the code point iterator
+     */
+    public CodePointIterator skipCrLf() {
+        return skip(value -> value == '\n' || value == '\r');
+    }
+
+    /**
+     * Get a sub-iterator that removes code points based on a <code>predicate</code>.
+     *
+     * @param predicate a {@link IntPredicate} that evaluates the code points that should be skipper. Returning true from the predicate
+     * indicates that the code point must be skipped.
+     * @return the code point iterator
+     */
+    public CodePointIterator skip(IntPredicate predicate) {
+        if (!hasNext()) {
+            return EMPTY;
+        }
+
+        return new CodePointIterator() {
+            public boolean hasNext() {
+                return CodePointIterator.this.hasNext() && !skip(peekNext());
+            }
+
+            public boolean hasPrev() {
+                return CodePointIterator.this.hasPrev() && !skip(peekPrev());
+            }
+
+            public int next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+
+                return CodePointIterator.this.next();
+            }
+
+            public int peekNext() throws NoSuchElementException {
+                if (!CodePointIterator.this.hasNext()) {
+                    throw new NoSuchElementException();
+                }
+
+                int next = seekNext(CodePointIterator.this.peekNext());
+
+                if (!skip(next)) {
+                    return next;
+                }
+
+                return next;
+            }
+
+            private int seekNext(int next) throws NoSuchElementException {
+                if (!CodePointIterator.this.hasNext()) {
+                    return next;
+                }
+
+                next = CodePointIterator.this.next();
+
+                if (skip(next)) {
+                    return seekNext(next);
```

Comment:
OK. Just let me know if you feel that we should fix that now. Otherwise, I'll create a JIRA to do that later.